### PR TITLE
Fix tslint and tsc warnings

### PR DESCRIPTION
**Pull request by topaxi**
Fixes all but one tslint and tsc warning.

```typescript
private PlayerComponent#isCurrentAudioFile()
```
seems unused.

### CODE REVIEW

**Comment by topaxi** [comment]
The dev build does not seem to catch some typescript errors... weird, looking into it.

File: src/app/archive/broadcasts/broadcasts_date.component.ts [hunk 1]
```diff
@@ -38,17 +38,17 @@ export class BroadcastsDateComponent {
     this.dateWithTimeSub = paramsObservable
       .subscribe(params => this.dateWithTime = this.getDateWithTime(params));
     this.broadcastsSub = dateObservable
-      .distinctUntilChanged(null, date => date.getTime())
-      .do(_ => this.loading = true)
+      .distinctUntilChanged(null, (date: Date) => date.getTime())
```
Comment (topaxi) [inline src/app/archive/broadcasts/broadcasts_date.component.ts:41]: Unable to find documentation about this `Observable#distinctUntilChanged()` method signature.

Official doc says it only has one param: http://reactivex.io/rxjs/class/es6/Observable.js~Observable.html#instance-method-distinctUntilChanged

File: src/app/archive/broadcasts/broadcasts_show.component.ts [hunk 2]
```diff
@@ -127,20 +128,20 @@ export class BroadcastsShowComponent {
   private broadcastMoreObservable(): Observable<CrudList<BroadcastModel>> {
     return this.fetchMore
       .withLatestFrom(this.broadcastList, (_, list) => list)
-      .distinctUntilChanged(null, list => list.links.next)
-      .do(_ => this.fetchingMore = true)
-      .switchMap(list =>
+      .distinctUntilChanged(null, (list: CrudList<BroadcastModel>) =>
```
Comment (topaxi) [inline src/app/archive/broadcasts/broadcasts_show.component.ts:131]: Unable to find documentation about this `Observable#distinctUntilChanged()` method signature.

Official doc says it only has one param: http://reactivex.io/rxjs/class/es6/Observable.js~Observable.html#instance-method-distinctUntilChanged

**Comment by topaxi** [comment]
Build fails due to the noted observable distinctUntilChanged _and_ node-sass native module failure.
Native module rebuild is addressed in #6

**Comment by codez** [comment]
Great, thanks a lot for all those improvements!

**Pull request merged**